flask_app/FlaskApp.py

```diff
@@ -1,16 +1,152 @@
 
 
 class FlaskAppHandler(metaclass=Singleton):
     def __init__(self):
         self.app = Flask(__name__, instance_relative_config=False)
         self.app.config.from_object(FlaskConfig)
         self.db = SQLAlchemyHandler(self.app)
+        self.azure_handler = AzureBlobContainerHandler()
+        self.azure_handler.create_container()
 
     def init_database(self):
+        """
+        Initiate database clear data models and create them
+        """
         with self.app.app_context():
             self.db.clear_data_models()
             self.db.create_data_models()
+
+    def run(self):
+        self.app.add_url_rule('/upload_video', view_func=self.upload_video, methods=['POST'])
+        self.app.run()
+
+    async def upload_video(self):
+        """
+        Get local video path upload to azure extract it to frames and metadata, create models and upload it to database.
+        :return: server code.
+        """
+        path = request.json['video_path']
+        Logger.logger.info(f'Start to handle path: {path}')
+
+        video_file_name = os.path.basename(path)
+        image_list = frame_utils.extract_video_to_frames(path)
+        thread_pool = ThreadPoolExecutor(max_workers=3)
+
+        Logger.logger.info(f'Start to upload video for path: {path}')
+        upload_video_task = asyncio.create_task(self.upload_binary_file_to_azure(path, video_file_name, 'videos/'))
+
+        Logger.logger.info(f'Start to upload images for path: {path}')
+        self.upload_images_to_azure(image_list, video_file_name, thread_pool)
+
+        Logger.logger.info(f'Start create and insert video frames and metadata to database, video path: {path}')
+        self.create_and_insert_to_database_video_metadata_frame_models(path, image_list, video_file_name)
+
+        await upload_video_task
+        thread_pool.shutdown(wait=True)
+
+        Logger.logger.info(f'finish to upload video for path: {path}')
+        Logger.logger.info(f'finish to upload images for path: {path}')
+        Logger.logger.info(f'Finish to insert video and frames to database, video path: {path}')
+        Logger.logger.info(f'Finish to handle path: {path}')
+
+        return json.dumps({'success': True}), 200, {'ContentType': 'application/json'}
+
+    def upload_images_to_azure(self, images: list, video_file_name: str, thread_pool: ThreadPoolExecutor):
+        """
+        Upload images to azure in background with thread pool
+        :param images: List of images
+        :param video_file_name: Video file name
+        :param thread_pool: Thread pool
+        """
+        for index, image in enumerate(images):
+            _, img_encode = cv2.imencode('.jpg', image)
+            img_bytes = img_encode.tobytes()
+            thread_pool.submit(self.azure_handler.upload_file, f'frame{index}.jpg',
+                               img_bytes, os.path.join('frames', video_file_name))
+
+    def create_and_insert_to_database_video_metadata_frame_models(self,
```

Comment:
why having this function here and not in a new script that is designed for inserting data ?